src/utils/common.py

```diff
@@ -234,7 +234,9 @@ def check_notifications_record(notifications, records):
             continue
 
         r_values = record[a]['values']
-        n_values = [e[a]['value'] for e in notifications]
+        # collect values for the attribute a from the entities in the notification
+        # if an entity does not have a value for the attribute use None
+        n_values = [e.get( a, { 'value': None } )['value'] for e in notifications]
```

Comment:
so basically, this is the place where "none" is entered if an attribute value is missing.

i think a specific test to validate this change is needed. also this change may be better to be in a separate pr
@taliaga @c0c0n3 ?

Reply:
As far as I can tell, the setting of missing attribute values to `None` happens in the `reporter`

https://github.com/smartsdk/ngsi-timeseries-api/pull/122/files/a78eb256fbe307599ce1c6fef2e0f9623f9f3c50#diff-f796da60daee45177c252e52fc78b4aaR96:

Again, this is a change we brought in with PR #122---see my comment above.
On the other hand this block of code @ohylli changed is part of the `check_notifications_record` function which is only called from test code, so if all tests pass, this must be a good change, right? However, in principle this line shouldn't be needed, again because of PR #122. So I suggest rolling back the changes to this file, unless doing so causes some tests to fail, in which case we'd have to figure out why.

Reply:
This modification is required by the new Test_insert_same_type_entities_with_different_attrs test I added where I use this method. This test uses the new test data I added to src/conftest. As I said in my previous comment about my modification to _preprocess_values the situation with this modification is also different from pr #122 since here the attribute is completely missing instead of only the value missing.

Reply:
I understand this is a needed change because now with this PR the symmetry of the entities (in terms of set of attributes) is no longer hold, because now an `a` (attribute) of entity `e1` (entity) might not exist in another `e2`.

The catch I mentioned earlier is, under this assumption of symmetry, the `check_notifications_record` method has been taking the set of attributes from the first record... https://github.com/smartsdk/ngsi-timeseries-api/blob/f639c6c3a5c4d774b59db8e16716b0616ca747ab/src/utils/common.py#L221. To make sure you test all attributes (their values and Nones), you should instead take the complete set of attributes (considering all records).

Thanks in advance for your patience with our comments. Since this PR is changing a critical assumption, we want to make sure we don't oversee places where this symmetry break could cause more problems (specially those that do not break tests)

On a side note,
- Maybe slightly cleaner: ```n_values = [e[a]['value'] if a in e else None for e in notifications]```
- No need for a separate PR IMO

Reply:
Is this required for this pull request since all tests including my new one gives one record for this. check_notifications_record function also has an assert for checking that there is only one record.

Reply:
Of course if we end up creating more tests for this pr then  check_notifications_record  probably will require this modification.

Reply:
no problem, we can take care of that later.